gpii/node_modules/preferencesServer/test/preferencesServerTests.js

```diff
@@ -595,14 +595,14 @@ gpii.tests.preferencesServer.get.failFixtures = [
         name: "GET: Retrieving preferences by a nonexistent GPII key returns an error",
         gpiiKey: "non-existent-gpii-key",
         url: "%gpiiKey",
-        expectedStatusCode: 500,
+        expectedStatusCode: 404,
         expected: {
             isError: true,
             message: "GPII key \"non-existent-gpii-key\" does not exist"
         }
     }, {
         name: "GET: Not providing GPII key returns an error",
-        expectedStatusCode: 500,
+        expectedStatusCode: 404,
```

Comment:
Not a big deal but I think maybe this scenario should return 400:

> 400 Bad Request
> the server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).
--https://tools.ietf.org/html/rfc7231#section-6.5.1

Unlikely to matter in practice so not sure if this is worth addressing; 404 is clearer than 500, anyway :).

Reply:
But isn't 404 (Not Found) clearer in this case when a GPII key is not found? 

The spec you referred says: `The 404 (Not Found) status code indicates that the origin server did not find a current representation for the target resource or is not willing to disclose that one exists.`

Reply:
Yes, I agree 404 is the best response when a key is not found (e.g. the scenario starting on line 595).

This scenario is where there is no key in the request at all. To me, that is a malformed request ("give me the key" "which key?" "").

Reply:
Thanks for the clarification, @mrtyler, it does make sense. 

After more testing, I found when no key is provided in a http request, this request will not reach GET /preferences/:gpiiKey endpoint handler. It will be filtered out by the server and the sender will receive a 404 response with an error "Cannot GET /preferences/".

This test case is incorrect as its attempt is to provide an `undefined` GPII key but this special data type is transformed into the string "undefined" when being appended to the request. This test case ends up testing the retrieval of a GPII key named "undefined" and becomes a duplicate of the test above. I've removed it in the new commit.